src/back/wgsl/writer.rs

```diff
@@ -530,15 +688,74 @@ impl<W: Write> Writer<W> {
             Expression::Compose { ty, ref components } => {
                 self.write_type(&module, ty)?;
                 write!(self.out, "(")?;
-                self.write_slice(components, |this, _, arg| {
-                    this.write_expr(&module, *arg, func_ctx)
-                })?;
+                // !spv-in specific notes!
+                // WGSL does not support all SPIR-V builtins and we should skip it in generated shaders.
+                // We already skip them when we generate struct type.
+                // Now we need to find components that used struct with ignored builtins.
+
+                // So, why we can't just return the error to a user?
+                // We can, but otherwise, we can't generate WGSL shader from any glslang SPIR-V shaders.
+                // glslang generates gl_PerVertex struct with gl_CullDistance, gl_ClipDistance and gl_PointSize builtin inside by default.
+                // All of them are not supported by WGSL.
+
+                // We need to copy components to another vec because we don't know which of them we should write.
+                let mut components_to_write = Vec::with_capacity(components.len());
+                for component in components {
+                    let mut skip_component = false;
+                    if let Expression::Load { pointer } = func_ctx.expressions[*component] {
+                        if let Expression::AccessIndex {
+                            base,
+                            index: access_index,
+                        } = func_ctx.expressions[pointer]
+                        {
+                            let base_ty_res = &func_ctx.info[base].ty;
+                            let resolved = base_ty_res.inner_with(&module.types);
+                            if let TypeInner::Pointer {
+                                base: pointer_base_handle,
+                                ..
+                            } = *resolved
+                            {
+                                // Let's check that we try to access a struct member with unsupported built-in and skip it.
+                                if let TypeInner::Struct { ref members, .. } =
+                                    module.types[pointer_base_handle].inner
+                                {
+                                    if let Some(Binding::BuiltIn(builtin)) =
+                                        members[access_index as usize].binding
+                                    {
+                                        if builtin_str(builtin).is_none() {
+                                            // glslang why you did this with us...
+                                            log::warn!(
+                                                "Skip component with unsupported builtin {:?}",
+                                                builtin
+                                            );
+                                            skip_component = true;
```

Comment:
this can be just `continue`, right?

Reply:
Unfortunately, no. We need to go through all this if chain to be sure that we find right struct and it member.

> So we no longer need to use the assignable_global thing?

Yes! I removed this changes from PR.